pkg/flow/internal/controller/loader.go

```diff
@@ -310,9 +334,31 @@ func (l *Loader) populateServiceNodes(g *dag.Graph) diag.Diagnostics {
 			node = NewServiceNode(l.host, svc)
 		}
 
+		node.UpdateBlock(nil) // Reset configuration to nil.
 		g.Add(node)
 	}
 
+	// Now, assign blocks to services.
+	for _, block := range serviceBlocks {
+		blockID := BlockComponentID(block).String()
+		node := g.GetByID(blockID).(*ServiceNode)
+
+		// Blocks assigned to services are reset to nil in the previous loop. If
+		// they're non-nil now, it means a service block incorrectly was provided
+		// twice.
+		if node.Block() != nil {
```

Comment:
Given the above code, I think that it would always be nil here, right?

Reply:
No, it'll be non-nil in this case:

```river
SERVICE_NAME {

}

SERVICE_NAME {

}
```

I'll try to clarify the comment to explain that this is the scenario I'm talking about.